scripts/zones/Port_Bastok/npcs/Oggbi.lua

```diff
@@ -1,56 +1,57 @@
 -----------------------------------
--- Area: Port Bastok
---  NPC: Oggbi
--- Starts and Finishes: Ghosts of the Past, The First Meeting
--- @zone 236
+-- Area: Port Bastok (236)
+-- NPC: Oggbi
+-- Starts and Finishes: Ghosts of the Past, The First Meeting, The Walls of Your Mind
+-- !zone 236
 -- !pos -159 -7 5
 -----------------------------------
 package.loaded["scripts/zones/Port_Bastok/TextIDs"] = nil;
 -----------------------------------
 require("scripts/globals/settings");
 require("scripts/globals/keyitems");
 require("scripts/globals/quests");
+require("scripts/globals/wsquest");
 require("scripts/zones/Port_Bastok/TextIDs");
 -----------------------------------
 
+local wsQuest = dsp.wsquest.theWallsOfYourMind;
+
 function onTrade(player,npc,trade)
+    local wsQuestEvent = getWsQuestTradeEvent(wsQuest, player, trade);
 
     if (player:getQuestStatus(BASTOK,GHOSTS_OF_THE_PAST) == QUEST_ACCEPTED) then
         if (trade:hasItemQty(13122,1) and trade:getItemCount() == 1) then -- Trade Miner's Pendant
             player:startEvent(232); -- Finish Quest "Ghosts of the Past"
         end
-    end
+    end;
 
+    if (wsQuestEvent ~= nil) then
+        player:startEvent(wsQuestEvent);
+    end;
 end;
 
 function onTrigger(player,npc)
-
-    ghostsOfThePast = player:getQuestStatus(BASTOK,GHOSTS_OF_THE_PAST);
-    theFirstMeeting = player:getQuestStatus(BASTOK,THE_FIRST_MEETING);
-    mLvl = player:getMainLvl();
-    mJob = player:getMainJob();
+    local wsQuestEvent = getWsQuestTriggerEvent(wsQuest, player);
+    local ghostsOfThePast = player:getQuestStatus(BASTOK,GHOSTS_OF_THE_PAST);
+    local theFirstMeeting = player:getQuestStatus(BASTOK,THE_FIRST_MEETING);
+    local mLvl = player:getMainLvl();
+    local mJob = player:getMainJob();
 
     if (ghostsOfThePast == QUEST_AVAILABLE and mJob == 2 and mLvl >= 40) then
         player:startEvent(231); -- Start Quest "Ghosts of the Past"
     elseif (ghostsOfThePast == QUEST_COMPLETED and player:needToZone() == false and theFirstMeeting == QUEST_AVAILABLE and mJob == 2 and mLvl >= 50) then
         player:startEvent(233); -- Start Quest "The First Meeting"
     elseif (player:hasKeyItem(dsp.ki.LETTER_FROM_DALZAKK) and player:hasKeyItem(dsp.ki.SANDORIAN_MARTIAL_ARTS_SCROLL)) then
         player:startEvent(234); -- Finish Quest "The First Meeting"
+    elseif (wsQuestEvent ~= nil) then
```

Comment:
Kind of pedantic, but based on wiki guide for this quest suggest it WSNM quests are at the top of the food chain, which for us is just making sure it's the first condition in the if statement for now.

I kind of assume if this is true, it'd be the case for all other WSNM quests.

http://ffxiclopedia.wikia.com/wiki/Ghosts_of_the_Past
```Note: If you have hand-to-hand skill above 250 and he offers you the Knuckles of Trials for the quest The Walls of Your Mind you will not be able to start this quest until you've completed the trial.```